src/core/qgscoordinatereferencesystem.cpp

```diff
@@ -1417,8 +1417,8 @@ bool QgsCoordinateReferenceSystem::readXml( const QDomNode &node )
   }
   else
   {
-    // Return default CRS if none was found in the XML.
-    createFromId( GEOCRS_ID, InternalCrsId );
+    // Return empty CRS if none was found in the XML.
+    operator=( QgsCoordinateReferenceSystem::QgsCoordinateReferenceSystem( ) );
```

Comment:
This should avoid some unnecessary overhead here:

    d = new QgsCoordinateReferenceSystemPrivate();